### PR TITLE
Do not call _doShow() if the view is already in ATTACHED_SHOWN state

### LINKED ISSUE #1325: Dev warning about invalid view state for _doShow()
Currently, if a view is in the `ATTACHED_SHOWN` state, it'll fire off a warning if `_doShow()` is getting called again (by `_isVisibleDidChange`).

However, if `isVisible` is a computed property, it's possible that it could change from `true` to `true`. This seems like a valid case. However, `_isVisibleDidChange` could call `_doShow` twice, causing a dev warning the second time as the view is already in `ATTACHED_SHOWN` state.

There are two possible fixes:
* `_isVisibleDidChange` here should not call `_doShow` in this case
* `_doShow` doesn't display a warning if being called from `ATTACHED_SHOWN` state

The first approach seems the more correct solution to me

### PR DESCRIPTION
**Pull request by jameschao**
Fixes #1325.

### CODE REVIEW

**Comment by jameschao** [comment]
Hmmm, the only failed tests are in websockets:
```
[305/474] /static/sproutcore/ajax/en/current/tests/system/websocket.html (errors): Completed 5 tests in 240 msecs. 9 total assertions: 4 passed 3 failed 2 errors (with unhandled errors)
SC.WebSocket module: Basic Requirements
  webSocket should be supported: failed
SC.WebSocket module: Test nofitications
  Test timed out: failed
SC.WebSocket module: Test delegate handling
  Test timed out: failed
SC.WebSocket module: Test queue
  Died on test #2: TypeError: 'undefined' is not an object (evaluating 'this.onOpen.apply'): errors
SC.WebSocket module: Test authentification
  Died on test #2: TypeError: 'undefined' is not an object (evaluating 'this.onOpen.apply'): errors
```

Is there something I need to do to get these tests passing, or is this a known issue?

**Comment by publickeating** [comment]
I broke those tests recently while cleaning up WebSocket. Don't worry about it, I'll fix them right away.

File: frameworks/core_foundation/views/view/statechart.js [hunk 1]
```diff
@@ -1511,7 +1511,10 @@ SC.CoreView.reopen(
   */
   _isVisibleDidChange: function () {
     if (this.get('isVisible')) {
-      this._doShow();
+      // show the view if it's not being shown already
+      if (this.get('viewState') !== SC.CoreView.ATTACHED_SHOWN) {
```
Comment (publickeating) [inline frameworks/core_foundation/views/view/statechart.js]: There are actually a few different "shown" states. Can you make this into,

    if (!(this.get('viewState') & SC.CoreView.IS_SHOWN)) {

?

Comment (jameschao) [inline frameworks/core_foundation/views/view/statechart.js]: Got it. Will make the change and update tests.

**Comment by jameschao** [comment]
@publickeating, I made the changed like you suggested, but I had to add an exception for `ATTACHED_HIDING` as in that state `_doShow()` actually needs to set off a new transition back to the attached shown state.

**Comment by publickeating** [comment]
Thanks for this change. Merged it in.

I've struggled with the delineation of view states. "Hiding" is an example where it might be better for that not to be part of the shown group, which would remove that workaround you encountered (good catch, btw). Another case would be "building out". I'll consider it the next time I dig into the view states.

**Pull request merged**